### DISCUSSION TITLE
[Help] Is there a way to get the title of the current file as a variable?

### DISCUSSION THREAD

**Discussion by chikcne** [zk-org/zk-nvim#133] [Help] [answered]
### Verify

- [X] I searched the existing discussions for help

### How can we help you?

I want to display the title of the file displayed by the buffer as the winbar text. Is there a way to grab the title of the file from `zk-nvim`? I notice telescope grabs it from the `notes.title` in its `entry_maker` but it seems like this table is obtained from the LSP via something like `zk list`, which means it wouldn't be suitable for this use.

**Comment by mickael-menu** [comment, marked as answer]
You can try this in lua:

```lua
require('zk.api').list(
  nil,
  { hrefs = { vim.fn.expand('%') }, select = {"title"} },
  function(err, res)
    print(res[1]["title"])
  end
)
```

Or if you want to use an ex command, go straight to the CLI tool:

```viml
!zk list % -q --format {{title}}
```

**Comment by chikcne** [comment]
Thanks for the reply, for the Lua script, as I understand it the `zk-nvim` implementation of `zk.list` does not actually return the value,  rather it runs a callback function on it. (I don't actually want the value printed, I need to store it as a variable to be used elsewhere). Is it possible to return the value?

In the meantime something like `vim.b` works to store the value in the buffer (although this still has limitations):
```
title = require('zk.api').list(
  nil,
  { hrefs = { vim.fn.expand('%') }, select = {"title"} },
  function(err, res)
    vim.b.title = res[1]["title"]
  end
)
```

**Reply by mickael-menu** [reply]
Yes this API is asynchronous, there's no way to run it synchronously as the native Lua LSP client is asynchronous.

If you really need it synchronous, I would resort to using the CLI tool directly. But ideally you would just update the winbar asynchronously.
